src/data/battler-tags/damage-protected-tag.ts

```diff
@@ -0,0 +1,14 @@
+import { ProtectedTag } from "#app/data/battler-tags/protected-tag";
+import type { Move } from "#app/data/moves/move";
+import type { Pokemon } from "#app/field/pokemon";
+import { MoveCategory } from "#enums/move-category";
+
+/** Base class for `BattlerTag`s that block damaging moves but not status moves */
+export class DamageProtectedTag extends ProtectedTag {
```

Comment:
seems like this could also be abstract
```suggestion
export abstract class DamageProtectedTag extends ProtectedTag {
```